extensions/amp-story/1.0/variable-service.js

```diff
@@ -86,27 +91,28 @@ export class AmpStoryVariableService {
           totalPages,
           previousPageId,
         } = stateChangeEvent.value;
-        this.variables_[Variable.STORY_PAGE_INDEX] = pageIndex;
-        this.variables_[Variable.STORY_PAGE_ID] = pageId;
-        this.variables_[Variable.STORY_PROGRESS] = storyProgress;
-        this.variables_[Variable.STORY_PAGE_COUNT] = totalPages;
-        this.variables_[Variable.STORY_PREVIOUS_PAGE_ID] = previousPageId;
+
+        if (!this.storeService_.get(StateProperty.AD_STATE)) {
+          this.variables_[AnalyticsVariable.STORY_PROGRESS] = storyProgress;
```

Comment:
What do you think about not passing a wrong value to the variable service in the first place?

Reply:
I'll address this after #23110

Reply:
Sorry, just catching up on these. Do we know that pubs never want these pings from ads?

Reply:
Actually this will fix it, right now the `STORY_PROGRESS` variable was `1` (end of story), even when the ad was in the middle of the story.
No trigger is changed, but the variable is fixed

Reply:
Got it. SGTM thanks